File: java/src/main/java/org/example/algorithms/hashmap/TwoSum.java
package org.example.algorithms.hashmap;

import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    // https://leetcode.com/problems/two-sum/description/
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hasher = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            if (hasher.containsKey(nums[i])) {
                return new int[]{hasher.get(nums[i]), i};
            }

            hasher.put(target - nums[i], i);
        }

        return new int[0];
    }
}

File: go/algorithms/hashmap/two_sum.go
package hashmap

func twoSum(nums []int, target int) []int {
	hasher := make(map[int]int)

	for i, value := range nums {
		if v, ok := hasher[value]; ok {
			return []int{v, i}
		}
		hasher[target-value] = i
	}

	return []int{}
}
